spyder_terminal/widgets/terminalgui.py

```diff
@@ -240,7 +244,12 @@ def copy(self):
 
     def paste(self):
         """Paste unicode text into terminal."""
-        self.triggerPageAction(QWebEnginePage.Paste)
+        clipboard = QApplication.clipboard()
+        text = str(clipboard.text())
+        if len(text.splitlines()) > 1:
+            eol_chars = os.linesep
+            text = eol_chars.join((text + eol_chars).splitlines())
+        self.eval_javascript('pasteText({})'.format(repr(text)))
```

Comment:
Please add a regression test that verifies that multi-line pasting is working, so that additional changes won't break it in the future.